test/e2e/installplan_e2e_test.go

```diff
@@ -412,17 +430,16 @@ func TestCreateInstallPlanWithCSVsAcrossMultipleCatalogSources(t *testing.T) {
 		},
 	}
 
-	cleanup, err := decorateCommonAndCreateInstallPlan(crc, testNamespace, installPlan)
+	cleanup, err := decorateCommonAndCreateInstallPlan(t, c, crc, testNamespace, installPlan)
 	require.NoError(t, err)
 	t.Logf("Install plan %s created", installPlan.GetName())
 	defer cleanup()
 
-	// Wait for InstallPlan to be status: Complete before checking resource presence
-	fetchedInstallPlan, err := fetchInstallPlan(t, crc, installPlan.GetName(), installPlanCompleteChecker)
+	// Wait for InstallPlan to be status: Complete or Failed before checking resource presence
+	fetchedInstallPlan, err := fetchInstallPlan(t, crc, installPlan.GetName(), installPlanCompleteOrFailedChecker)
```

Comment:
now that we're doing permutations of state checks, might be time to go ahead and write builder: `func installPlanStatusChecker(statuses ...InstallPlanStatus) checkerFunc`